util/rbac/rbac.go

```diff
@@ -181,28 +219,28 @@ func (e *Enforcer) EnforceRuntimePolicy(policy string, rvals ...interface{}) boo
 // CreateEnforcerWithRuntimePolicy creates an enforcer with a policy defined at run-time which augments the built-in and
 // user-defined policy. This allows any explicit denies of the built-in, and user-defined policies
 // to override the run-time policy. Runs normal enforcement if run-time policy is empty.
-func (e *Enforcer) CreateEnforcerWithRuntimePolicy(policy string) *casbin.Enforcer {
-	var enf *casbin.Enforcer
+func (e *Enforcer) CreateEnforcerWithRuntimePolicy(policy string) WrapperEnforcer {
+	var enf WrapperEnforcer
 	var err error
 	if policy == "" {
-		enf = e.Enforcer
+		enf = e.wrapperEnforcer
 	} else {
 		enf, err = newEnforcerSafe(newBuiltInModel(), newAdapter(e.adapter.builtinPolicy, e.adapter.userDefinedPolicy, policy))
```

Comment:
If the project has roles with policies then the cache is not used. We should support this use case as well and use cached enforces with project with policies.